packages/flutter/test/material/calendar_date_picker_test.dart

```diff
@@ -0,0 +1,835 @@
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowLeft);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowLeft);
+        await tester.pumpAndSettle();
+
+        // Activate it
+        await tester.sendKeyEvent(LogicalKeyboardKey.space);
+        await tester.pumpAndSettle();
+
+        // Should have selected Jan 18
+        expect(selectedDate, DateTime(2016, DateTime.january, 18));
+      });
+
+      testWidgets('Navigating with arrow keys scrolls months', (
+          WidgetTester tester) async {
+        DateTime? selectedDate;
+        await tester.pumpWidget(calendarDatePicker(
+          onDateChanged: (DateTime date) => selectedDate = date,
+        ));
+        // Navigate to the grid
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.pumpAndSettle();
+
+        // Navigate from Jan 15 to Dec 31 with arrow keys
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowLeft);
+        await tester.pumpAndSettle();
+
+        // Should have scrolled to Dec 2015
+        expect(find.text('December 2015'), findsOneWidget);
+
+        // Navigate from Dec 31 to Nov 26 with arrow keys
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.sendKeyEvent(LogicalKeyboardKey.arrowUp);
+        await tester.pumpAndSettle();
+
+        // Should have scrolled to Nov 2015
+        expect(find.text('November 2015'), findsOneWidget);
+
+        // Activate it
+        await tester.sendKeyEvent(LogicalKeyboardKey.space);
+        await tester.pumpAndSettle();
+
+        // Should have selected Jan 18
+        expect(selectedDate, DateTime(2015, DateTime.november, 26));
+      });
+
+      testWidgets(
+          'RTL text direction reverses the horizontal arrow key navigation', (
+          WidgetTester tester) async {
+        DateTime? selectedDate;
+        await tester.pumpWidget(calendarDatePicker(
+          onDateChanged: (DateTime date) => selectedDate = date,
+          textDirection: TextDirection.rtl,
+        ));
+        // Navigate to the grid
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+        await tester.pumpAndSettle();
+
+        // Navigate from Jan 15 to 19 with arrow keys
```

Comment:
I think all comments in this file should end with a period
```suggestion
        // Navigate from Jan 15 to 19 with arrow keys.
```